DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -426,4 +442,28 @@ function Get-ChocoInstalledPackage ($action) {
     Return $res
 }
 
+function Get-ChocoVersion ($action) {
+    $chocoInstallCache = Join-Path -Path $env:ChocolateyInstall -ChildPath 'cache'
+    if ( -not (Test-Path $chocoInstallCache)){
+        New-Item -Name 'cache' -Path $env:ChocolateyInstall -ItemType Directory | Out-Null
+    }
+    $chocoVersion = Join-Path -Path $chocoInstallCache -ChildPath 'ChocoVersion.xml'
+
+    if ($action -eq 'Purge') {
+        Remove-Item $chocoVersion -Force
+        $res = $true
+    } else {
+        $cacheSec = (Get-Date).AddSeconds('-60')
+        if ( $cacheSec -lt (Get-Item $chocoVersion -ErrorAction SilentlyContinue).LastWriteTime ) {
+            $res = Import-Clixml $chocoVersion
+        } else {
+            $cmd = choco -v
+            $res = [version]($cmd.Split('-')[0])

```

Comment:
Note that your commit message is incorrect. It's not `-beta`, it's really anything that comes with `-` attached to it. It's the concept of prereleases, which has been around since NuGet was first released and subsequently when Chocolatey was first released.

Reply:
It could be `-alpha.20171212`, or really any value.